docs/running-on-yarn.md

```diff
@@ -761,8 +761,27 @@ The following extra configuration options are available when the shuffle service
     NodeManagers where the Spark Shuffle Service is not running.
   </td>
 </tr>
+<tr>
+  <td><code>spark.yarn.shuffle.service.metrics.namespace</code></td>
+  <td><code>sparkShuffleService</code></td>
+  <td>
+    The namespace to use when emitting shuffle service metrics into Hadoop metrics2 system of the
+    NodeManager.
```

Comment:
Could you add some description about the limitation with old Hadoop versions (like 2.7.x)? Here or at Section `Running multiple versions of the Spark Shuffle Service`?

Reply:
For this section, everything will work as expected on Hadoop 2.7.x. The "Running multiple versions" section won't work on 2.7, but I already called out the supported YARN versions there. Can you let me know if there's anything else you think I should call out?

Reply:
I'm worrying about the situation some users try to use `Apache Spark distribution (with Hadoop 2.7)` at YARN 2.9+ cluster. Does it work?

Reply:
it looks like the name referenced by the node manager works with the Hadoop 2.9+ custom class loader, but I assume with Hadoop 2.7 it requires the spark_shuffle name ?  hence the spark.shuffle.service.name won't work unless you have recompiled the code and manually changed it.
Perhaps we just need to be more explicit in the config spark.shuffle.service.name that either references  the section running multiple versions of the Spark Shuffle Service or explicitly states supported in YARN 2.9+.     I assume this config with metrics doesn't matter as far as Hadoop version.
Also did we explicitly test with Hadoop 2.7 and the case @dongjoon-hyun brings up?

Reply:
> it looks like the name referenced by the node manager works with the Hadoop 2.9+ custom class loader, but I assume with Hadoop 2.7 it requires the spark_shuffle name ? hence the spark.shuffle.service.name won't work unless you have recompiled the code and manually changed it.

No, this is not correct. YARN ignores the hard-coded name on _all_ versions of YARN. Take a look at `AuxServices` on the 2.7.0 branch: 
https://github.com/apache/hadoop/blob/f95b390df2ca7d599f0ad82cf6e8d980469e7abb/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/AuxServices.java#L129-L136

`spark.shuffle.service.name` works fine on Hadoop 2.7, it is only the isolated classloader that won't work on older versions.

> I'm worrying about the situation some users try to use `Apache Spark distribution (with Hadoop 2.7)` at YARN 2.9+ cluster. Does it work?

I don't quite understand the concern here. Does my explanation above address your question? We haven't changed any of the interfaces used to interact with YARN, there should be no binary compatibility issues or anything of that sort. I can test whichever combination of `Spark Version + Hadoop Version Distribution` running on top of `Hadoop Version YARN` you like, but I am failing to see where the concern is / what you'd like me to look for.

Reply:
great, I'm glad it works with 2.7 as well, thanks for clarifying.  Yeah the concern was if it didn't work in 2.7 so I think you answered that.